configs/eslint/node/typescript/index.js

```diff
@@ -31,10 +32,7 @@ module.exports = {
     // TypeScript files
     {
       files: ['**/*.{cts,ts}'],
-      extends: [
-        'plugin:@typescript-eslint/eslint-recommended',
-        'plugin:@typescript-eslint/recommended',
-      ],
+      extends: ['plugin:@typescript-eslint/recommended-type-checked'],
```

Comment:
For simplicity, I didn't extend `'plugin:@typescript-eslint/stylistic-type-checked'`.

https://typescript-eslint.io/linting/configs